services/frontend-v3/src/components/competencyTable/CompetencyTable.jsx

```diff
@@ -0,0 +1,173 @@
+  const [loading, setLoading] = useState(true);
+  const [reset, setReset] = useState(false);
+  const [searchText, setSearchText] = useState("");
+  const [searchedColumn, setSearchedColumn] = useState("");
+  const [selectedRowKeys, setSelectedRowKeys] = useState([]);
+  const [size] = useState("large");
+
+  const { type } = props;
+
+  const getCompetencies = async () => {
+    try {
+      let results = await axios.get(
+        backendAddress + "api/admin/options/" + type
+      );
+      return results.data;
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const getDisplayType = (plural) => {
+    if (plural)
+      return props.intl.formatMessage({
+        id: "admin." + type + ".plural",
+        defaultMessage: type,
+      });
+
+    return props.intl.formatMessage({
+      id: "admin." + type + ".singular",
+      defaultMessage: type,
+    });
+  };
+
+  const handleSearch = (selectedKeys, confirm, dataIndex) => {
+    confirm();
+    setSearchText(selectedKeys[0]);
+    setSearchedColumn(dataIndex);
+  };
+
+  const handleReset = (clearFilters) => {
+    clearFilters();
+    setSearchText("");
+  };
+
+  const handleSubmitAdd = async (values) => {
+    try {
+      const url = backendAddress + "api/admin/options/" + type;
+
+      await axios.post(url, {
+        descriptionEn: values.addCompetencyEn,
+        descriptionFr: values.addCompetencyFr,
+        categoryId: 22,
+      });
+
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const handleSubmitEdit = async (values, id) => {
+    try {
+      const url = backendAddress + "api/admin/options/" + type + "/" + id;
+
+      await axios.put(url, {
+        descriptionEn: values.editCompetencyEn,
+        descriptionFr: values.editCompetencyFr,
+      });
+
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const handleSubmitDelete = async () => {
+    try {
+      const url = backendAddress + "api/admin/delete/" + type;
+
+      await axios.post(url, { ids: selectedRowKeys });
+
+      setSelectedRowKeys([]);
+      setReset(true);
+    } catch (error) {
+      console.log(error);
+      return 0;
+    }
+  };
+
+  const rowSelection = {
+    onChange: (selectedRowKeys) => {
+      onSelectChange(selectedRowKeys);
+    },
+  };
+
+  const onSelectChange = (selectedRowKeys) => {
+    // console.log("selectedRowKeys changed: ", selectedRowKeys);
+    setSelectedRowKeys(selectedRowKeys);
+  };
+
+  const convertToViewableInformation = () => {
+    const description =
+      props.intl.formatMessage({ id: "language.code" }) === "en"
+        ? "descriptionEn"
+        : "descriptionFr";
+
+    let allCompetencies = _.sortBy(data, description);
+
+    for (let i = 0; i < allCompetencies.length; i++) {
+      allCompetencies[i].key = allCompetencies[i].id;
+    }
+
+    return allCompetencies;
+  };
+
+  useEffect(() => {
```

Comment:
Generally its safer for hooks to be closer to the top of the file

Reply:
It's because they need to be created every time the function runs and if anything stops the function before a hook, it throws an error

Reply:
Fixed.